src/bosh_azure_cpi/lib/cloud/azure/restapi/vmss_rest.rb

```diff
@@ -0,0 +1,365 @@
+# frozen_string_literal: true
+
+module Bosh::AzureCloud
+  class AzureClient
+    # VMSS related starts
+
+    # Get the information for the vmss
+    #
+    # @param [String] resource_group_name  - Name of resource group.
+    # @param [String] vmss_name  - Name of the vmss
+    # @return [Hash]
+    #
+    # @See https://docs.microsoft.com/en-us/rest/api/compute/virtualmachinescalesets/get
+    #
+    def get_vmss_by_name(resource_group_name, vmss_name)
+      vmss = nil
+      result = _get_vmss_by_name(resource_group_name, vmss_name)
+      unless result.nil?
+        vmss = {}
+        vmss[:location] = result['location']
+        vmss[:sku] = {}
+        vmss[:sku][:name] = result['sku']['name']
+        vmss[:zones] = result['zones']
+      end
+      vmss
+    end
+
+    # Create one vmss
+    #
+    # @param [String] resource_group_name  - Name of resource group.
+    # @param [Hash] vm_params  - parameters for creating the vmss
+    # @param [Array] network_interfaces   - Network Interface Instances. network_interfaces[0] will be picked as the primary network and bound to public ip or load balancers.
+    # @return [Hash]
+    #
+    # @See https://docs.microsoft.com/en-us/rest/api/compute/virtualmachinescalesets/createorupdate
+    #
+    def create_vmss(resource_group_name, vm_params, network_interfaces)
+      url = rest_api_url(REST_API_PROVIDER_COMPUTE, REST_API_VIRTUAL_MACHINE_SCALE_SETS, resource_group_name: resource_group_name, name: vm_params[:vmss_name])
+      os_profile = {}
+      case vm_params[:os_type]
+      when 'linux'
+        os_profile['computerNamePrefix'] = vm_params[:vmss_name]
+        os_profile['adminUsername'] = vm_params[:ssh_username]
+        os_profile['linuxConfiguration'] = {
+          'disablePasswordAuthentication' => 'true',
+          'ssh' => {
+            'publicKeys' => [
+              {
+                'path'    => "/home/#{vm_params[:ssh_username]}/.ssh/authorized_keys",
+                'keyData' => vm_params[:ssh_cert_data]
+              }
+            ]
+          }
+        }
+      when 'windows'
+        # TODO: add support for the windows.
+        raise ArgumentError, "Unsupported os type: #{vm_params[:os_type]}"
+      else
+        raise ArgumentError, "Unsupported os type: #{vm_params[:os_type]}"
+      end
+
+      network_profile = {
+        'networkInterfaceConfigurations' => []
+      }
+      network_interfaces.each_with_index do |network, index|
+        nic = {
+          'name' => "nic-#{index}",
+          'properties' => {
+            'primary' => index.zero?,
+            'enableIPForwarding' => true,
+            'ipConfigurations' => [
+              {
+                'name' => "ipcfg-#{index}",
```

Comment:
The config name for one VM is "ipconfig#{index}". Shall we make it consistent?

https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/blob/5650dbf07dd19b354add9bec85c6f86deef5dde4/src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager_network.rb#L160

Reply:
good idea